src/app/store/teacher-subjects/teacher-subjects.selector.ts

```diff
@@ -0,0 +1,13 @@
+import { createSelector } from '@ngrx/store';
+
+import { State as AppState } from '../index';
+import { State as TeacherSubjectsState } from './teacher-subjects.reducer'
+
+export const selectSubjects = (state: AppState) => state.teacherSubjects;
+
+export const selectData = (state: TeacherSubjectsState) => state.data;
+
+export const selectAll = createSelector(
```

Comment:
good name convention is to use for callback function in selector the same name just with suffix `Pure` e.g.(`selectAllPure` in your case ) but it also worth to say what exactly `all` you are selecting e.g. `selectAllSubjects` and `selectAllSubjectsPure` for callback :)